packages/hebao_v1/contracts/stores/SecurityStore.sol

```diff
@@ -182,4 +266,62 @@ contract SecurityStore is DataStore
         wallets[wallet].inheritor = who;
         wallets[wallet].lastActive = uint128(now);
     }
-}
\ No newline at end of file
+
+    function cleanRemovedGuardians(address wallet)
+        private
+    {
+        Wallet storage w = wallets[wallet];
+
+        for (uint i = 0; i < w.guardians.length; i ++) {
+            Data.Guardian memory g = w.guardians[i];
+            if (isGuardianRemoved(wallet, g.addr)) {
+                Data.Guardian memory lastGuardian = w.guardians[w.guardians.length - 1];
+
+                if (g.addr != lastGuardian.addr) {
+                    w.guardians[i] = lastGuardian;
+                    w.guardianIdx[lastGuardian.addr] = i + 1;
+                }
+                w.guardians.pop();
+                delete w.guardianIdx[g.addr];
+                delete w.guardianValidSince[g.addr];
+                delete w.guardianValidUntil[g.addr];
+            }
+        }
```

Comment:
This only cleans up the guardians so it doesn't really matter, but this loop may skip cleaning up some guardians.

For example, if the guardians are like this: (valid, invalid, invalid) the loop will switch guardian[1] with guardian[2] and stop looping. But guardian[2] is also invalid so we'd have to check the guardian again at each index where there was a removal. You can do that by not incrementing `i` when a guardian was removed, but the more standard way to do this is to loop backwards over the list:
```
for (int i = int(w.guardians.length) - 1; i >= 0; i--) {
    Data.Guardian memory g = w.guardians[uint(i)];
    if (isGuardianRemoved(wallet, g.addr)) {
        Data.Guardian memory lastGuardian = w.guardians[w.guardians.length - 1];

        if (g.addr != lastGuardian.addr) {
            w.guardians[uint(i)] = lastGuardian;
            w.guardianIdx[lastGuardian.addr] = uint(i) + 1;
        }
        w.guardians.pop();
        delete w.guardianIdx[g.addr];
        delete w.guardianValidSince[g.addr];
        delete w.guardianValidUntil[g.addr];
    }
}
```

Reply:
updated